cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -493,6 +500,16 @@ else()
     )
 endif()
 
+if(DEFINED ENV{UTF8PROC_SOURCE_URL})
+  set(UTF8PROC_SOURCE_URL "$ENV{UTF8PROC_SOURCE_URL}")
+else()
+  set_urls(
+    UTF8PROC_SOURCE_URL
+    "https://github.com/JuliaStrings/utf8proc/archive/${ARROW_UTF8PROC_BUILD_VERSION}.tar.gz"
+    "https://github.com/ursa-labs/thirdparty/releases/download/latest/utf8proc-${ARROW_UTF8PROC_BUILD_VERSION}.tar.gz"
```

Comment:
This doesn't exist.
It seems that we don't need this backup URL. If github.com is down, both of them are failed.